tests/ConstMapSpecTest.hack

```diff
@@ -0,0 +1,73 @@
+/*
+ *  Copyright (c) 2016, Fred Emmott
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\TypeAssert;
+
+use namespace Facebook\TypeSpec;
+use type Facebook\TypeSpec\TypeSpec;
+
+final class ConstMapSpecTest extends TypeSpecTest<\ConstMap<arraykey, mixed>> {
+  <<__Override>>
+  public function getTypeSpec(): TypeSpec<\ConstMap<arraykey, int>> {
+    return TypeSpec\constmap(TypeSpec\arraykey(), TypeSpec\int());
+  }
+
+  <<__Override>>
+  public function getValidCoercions(): vec<(mixed, \ConstMap<arraykey, int>)> {
+    return vec[
+      tuple(Map {'foo' => 123}, Map {'foo' => 123}),
+      tuple(ImmMap {'foo' => 123}, ImmMap {'foo' => 123}),
+      tuple(dict['foo' => 123], ImmMap {'foo' => 123}),
+      tuple(dict[], ImmMap {}),
+      tuple(vec[123], ImmMap {0 => 123}),
+      tuple(vec['123'], ImmMap {0 => 123}),
+      tuple(keyset['123'], ImmMap {'123' => 123}),
+      tuple(varray[123], ImmMap {0 => 123}),
+      tuple(darray["123" => 123], ImmMap {'123' => 123}),
+    ];
+  }
+
+  <<__Override>>
+  public function getInvalidCoercions(): vec<(mixed)> {
+    return vec[
+      tuple(false),
+      tuple(123),
+      tuple(Map {'foo' => 'bar'}),
+    ];
+  }
+
+  <<__Override>>
+  public function getToStringExamples(
+  ): vec<(TypeSpec<\ConstMap<arraykey, mixed>>, string)> {
+    return vec[
+      tuple(
+        TypeSpec\constmap(TypeSpec\string(), TypeSpec\int()),
+        \ConstMap::class.'<string, int>',
```

Comment:
in tests, it might be better to use string literals instead of `::class` literals, to make sure we're testing that the namespaces get expanded correctly (or ignored for autoimported names, if that's what we want)

Reply:
with the possible exception of dict/vec/keyset, I definitely want to test that it really matches the runtime.

There might be value in testing both though